projects/aws/bottlerocket-bootstrap/pkg/kubeadm/utils.go

```diff
@@ -170,3 +173,60 @@ func unmarshalIntoMap(content []byte) (map[string]interface{}, error) {
 
 	return parsedMap, err
 }
+
+type EbsInitControl struct {
+	Command *exec.Cmd
+	Timeout <-chan time.Time
+}
+
+// startEbsInit starts the ebs-init script in the background
+// if marker file is present. Currenlty ebs-init is best-effort
+// and if it fails it won't prevent the instance from bootstrapping.
+// A 2 minutes timeout is added so that instance bootstrap time is
+// not inflated.
+func startEbsInit() *EbsInitControl {
+	if _, err := os.Stat(ebsInitMarker); err == nil {
+
+		if err = createEbsInitScript(); err != nil {
+			fmt.Printf("Failed to create ebs-init script")
+			return nil
+		}
+		cmd := exec.Command("bash", "-c", ebsInitScript)
+		if err = cmd.Start(); err != nil {
+			fmt.Printf("Failed starting ebs-init with %v \n", err)
+			return nil
+		}
+		fmt.Printf("Starting ebs-init \n")
+		ebsInitControl := &EbsInitControl{}
+		ebsInitControl.Timeout = time.After(2 * time.Minute)
+		ebsInitControl.Command = cmd
+		return ebsInitControl
+	} else {
+		fmt.Printf("Skipping ebs initialization \n")
+		return nil
+	}
+}
+
+// createEbsInitScript writes the ebs-init script contents to disk
+func createEbsInitScript() error {
+	fmt.Printf("Creating %s file \n", ebsInitScript)
+	contents := `
+#!/bin/bash
+# This script is meant to initialize EBS volumes
+# by reading all files on the filesystem.
+# Doing so will prevent potential blocked IO in
+# the event where the instance is running on 
+# Outpost and the Outpost gets disconnected
+# before EBS is synced from region.
+
+find /.bottlerocket -not \( -path "/.bottlerocket/rootfs/proc" -prune \) \
+-not \( -path "/.bottlerocket/rootfs/sys" -prune \) \
+-not \( -path "/.bottlerocket/rootfs/run" -prune \) -type f -print0 \
+| xargs -0 -I {} dd if={} of=/dev/null bs=1M &> /dev/null
+`
```

Comment:
Would it be possible to execute this in golang itself, without writing it to a shell script and executing it? Like use filepath.Walk for the find -type f and use the output to spin off a go-routine and run a read file asynchronously against the provided list. 

An example of non-blocking go-routine that runs till the context is cancelled or gets a Done signal from the go channel from inside the routine. https://github.com/aws/eks-anywhere-build-tooling/blob/main/projects/aws/bottlerocket-bootstrap/pkg/utils/waiters.go#L133

Reply:
This way we can have everything in go. Since this is best effort, we can start off this process in a separate routine at the start of the bootstrap process, and at the end, if we are waiting for this routine to finish we can check if its past the 2 min timeout since we started this routine, cancel it, and exit successfully.

Reply:
I'll give it a try and see how long it will take.